rtgui/filecatalog.cc

```diff
@@ -538,11 +541,44 @@ void FileCatalog::closeDir ()
 
 std::vector<Glib::ustring> FileCatalog::getFileList ()
 {
-
     std::vector<Glib::ustring> names;
-    Glib::RefPtr<Gio::File> dir = Gio::File::create_for_path (selectedDirectory);
-    safe_build_file_list (dir, names, selectedDirectory, &(options.parsedExtensions));
-// Issue 2406    std::sort (names.begin(), names.end());
+
+    std::set<Glib::ustring> extensions;
+    for (const auto& parsedExt : options.parsedExtensions) {
+        extensions.emplace (parsedExt.lowercase ());
+    }
+
+    try {
+
+        auto dir = Gio::File::create_for_path (selectedDirectory);
+
+        auto enumerator = dir->enumerate_children ();
```

Comment:
I suggest to use `auto enumerator = dir->enumerate_children ("standard::name");`